src/modules/persistence/services/pair.persistence.service.ts

```diff
@@ -122,4 +140,411 @@ export class PairPersistenceService {
 
         return pair;
     }
+
+    async populatePairs(): Promise<void> {
+        this.logger.info(`Starting ${this.populatePairs.name}`, {
+            context: PairPersistenceService.name,
+        });
+
+        const profiler = new PerformanceProfiler();
+        const pairsMetadata = await this.routerAbi.pairsMetadata();
+
+        for (const pairMeta of pairsMetadata) {
+            try {
+                await this.populatePairModel(pairMeta);
```

Comment:
Wouldn't be possible to return an operation and gather them all into a single `bulkUpdatePairs` call?